subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiDistributionResolver.groovy

```diff
@@ -96,7 +96,16 @@ class ToolingApiDistributionResolver {
         Dependency dep = resolutionServices.dependencyHandler.create(dependency)
         Configuration config = resolutionServices.configurationContainer.detachedConfiguration(dep)
         config.resolutionStrategy.disableDependencyVerification()
-        return config.files
+        int retryCount = 3
+        List<Throwable> exceptions = []
+        while (retryCount-- > 0) {
+            try {
+                return config.files
```

Comment:
❓ I think you need to create the detached configuration in this loop. If you don't, I suppose it would re-use the resolution result of the first try. Though I am not 100% sure.

Reply:
@ljacomet WDYT here?

Reply:
It is as Stefan said, the resolution result will be cached, whether it is a failure or a success. So you need to create a new configuration each time.

Reply:
Thanks for pointing out, how about now?

Reply:
LGTM!